File: pset6/credit/credit.py
from cs50 import get_int


CCNumber = get_int("Number: ")


def sum_digits(n):
    s = 0
    while n:
        s += n % 10
        n //= 10
    return s


"""First two digits and Length variables for later use"""
Length = len(str(CCNumber))
F2D = int(str(CCNumber)[:2])
FD = int(str(CCNumber)[0])


def LuhnsSum(Number):
    TOTALSUM = 0
    Length = len(str(Number))
    if (Length % 2) == 0:
        for i in range(Length//2):
            g = int((str(Number))[2*i])
            g = g * 2
            TOTALSUM += sum_digits(g)
        for i in range(Length//2):
            g = int((str(Number))[(2*i)+1])
            TOTALSUM += sum_digits(g)
    if (Length % 2) == 1:
        for i in range(Length//2):
            g = int((str(Number))[(2*i)+1])
            g = g * 2
            TOTALSUM += sum_digits(g)
        for i in range((Length//2)+1):
            g = int((str(Number))[2*i])
            TOTALSUM += sum_digits(g)

    if int((str(TOTALSUM))[-1]) == 0:
        return True
    else:
        return False


if(LuhnsSum(CCNumber)):
    if (Length == 15 and (F2D == 34 or F2D == 37)):
        print("AMEX")
    if (Length == 16 and (F2D == 51 or F2D == 52 or F2D == 53 or F2D == 54 or F2D == 55)):
        print("MASTERCARD")
    if ((Length == 13 or Length == 16) and (FD == 4)):
        print("VISA")

else:
    print("INVALID")

File: pset1/credit/credit.c
#include <cs50.h>
#include <stdio.h>
#include <math.h>

long long CNUMBER;

int main(void)
{do
{
CNUMBER = get_long("Please enter your credit card number here\n");
}
while (CNUMBER <= 0);

double kindalength = log10(CNUMBER);
int length = floor(kindalength) + 1;
int Digits[length];
int i = 0;
while (i < length)
    {int f = CNUMBER % 10;
    Digits[i] = f;
    CNUMBER = CNUMBER/10;
    i = i + 1;
    }
int b = 0;
int DoubleDigSum[length];
 while (b < length)
       {
    DoubleDigSum[b] = (Digits[b]*2);
    DoubleDigSum[b] = (DoubleDigSum[b] % 10) + DoubleDigSum[b]/10;
    b = b + 1;
    }
int totalsum;
if (length == 16)
      {
    totalsum = DoubleDigSum[1]+DoubleDigSum[3]+DoubleDigSum[5]+DoubleDigSum[7]+DoubleDigSum[9]+DoubleDigSum[11]+DoubleDigSum[13]+DoubleDigSum[15]+Digits[0]+Digits[2]+Digits[4]+Digits[6]+Digits[8]+Digits[10]+Digits[12]+Digits[14];
    }
if (length == 15)
      {
    totalsum = DoubleDigSum[1]+DoubleDigSum[3]+DoubleDigSum[5]+DoubleDigSum[7]+DoubleDigSum[9]+DoubleDigSum[11]+DoubleDigSum[13]+Digits[0]+Digits[2]+Digits[4]+Digits[6]+Digits[8]+Digits[10]+Digits[12]+Digits[14];
    }
if (length == 13)
      {
    totalsum = DoubleDigSum[1]+DoubleDigSum[3]+DoubleDigSum[5]+DoubleDigSum[7]+DoubleDigSum[9]+DoubleDigSum[11]+Digits[0]+Digits[2]+Digits[4]+Digits[6]+Digits[8]+Digits[10]+Digits[12];
    }
if (totalsum%10 == 0 && (length == 16 || length == 15 || length == 13))
{switch (Digits[length-1])
 {case 5:
  if(Digits[length-2] == 1 || Digits[length-2] == 2 || Digits[length-2] == 3 || Digits[length-2] == 4 || Digits[length-2] == 5)
  {
  printf("MASTERCARD\n");
  } else { printf("INVALID\n");
    }
  break;
  case 4:
  printf("VISA\n");
  break;
  case 3:
  if(Digits[length-2] == 4 || Digits[length-2] == 7)
  {
  printf("AMEX\n");
  } else { printf("INVALID\n");
    }
  break;
 }
 }
else 
  { printf("INVALID\n");
    }
 
}
